azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointImpl.java

```diff
@@ -0,0 +1,539 @@
+            public Void call(Void result) {
+                return result;
+            }
+        });
+    }
+
+    @Override
+    public CdnEndpointImpl refresh() {
+        EndpointInner inner = this.client.get(this.parent().resourceGroupName(),
+                this.parent().name(),
+                this.name());
+        this.setInner(inner);
+        this.customDomainList.clear();
+        this.deletedCustomDomainList.clear();
+        this.customDomainList.addAll(this.customDomainsClient.listByEndpoint(
+                this.parent().resourceGroupName(),
+                this.parent().name(),
+                this.name()));
+        return this;
+    }
+
+    @Override
+    public CdnProfileImpl attach() {
+        return this.parent().withEndpoint(this);
+    }
+
+    @Override
+    public String originHostHeader() {
+        return this.inner().originHostHeader();
+    }
+
+    @Override
+    public String originPath() {
+        return this.inner().originPath();
+    }
+
+    @Override
+    public List<String> contentTypesToCompress() {
+        return this.inner().contentTypesToCompress();
+    }
+
+    @Override
+    public boolean isCompressionEnabled() {
+        return this.inner().isCompressionEnabled();
+    }
+
+    @Override
+    public boolean isHttpAllowed() {
+        return this.inner().isHttpAllowed();
+    }
+
+    @Override
+    public boolean isHttpsAllowed() {
+        return this.inner().isHttpsAllowed();
+    }
+
+    @Override
+    public QueryStringCachingBehavior queryStringCachingBehavior() {
+        return this.inner().queryStringCachingBehavior();
+    }
+
+    @Override
+    public String optimizationType() {
+        return this.inner().optimizationType();
+    }
+
+    @Override
+    public List<GeoFilter> geoFilters() {
+        return this.inner().geoFilters();
+    }
+
+    @Override
+    public String hostName() {
+        return this.inner().hostName();
+    }
+
+    @Override
+    public EndpointResourceState resourceState() {
+        return this.inner().resourceState();
+    }
+
+    @Override
+    public String provisioningState() {
+        return this.inner().provisioningState();
+    }
+
+    @Override
+    public String originHostName() {
+        if (this.inner().origins() != null && !this.inner().origins().isEmpty()) {
+            return this.inner().origins().get(0).hostName();
+        }
+        return null;
+    }
+
+    @Override
+    public int httpPort() {
+        if (this.inner().origins() != null && !this.inner().origins().isEmpty()) {
+            return this.inner().origins().get(0).httpPort();
+        }
+        return 0;
+    }
+
+    @Override
+    public int httpsPort() {
+        if (this.inner().origins() != null && !this.inner().origins().isEmpty()) {
+            return this.inner().origins().get(0).httpsPort();
+        }
+        return 0;
+    }
+
+    @Override
+    public List<String> customDomains() {
+        return Lists.transform(this.customDomainList,
```

Comment:
doesn't this return a modifiable list? Might need to go through Collections.unmodifiableList(...). In general, unless they are PagedLists underneath, we return static lists as immutable, which is what unmodifiableList achieves..